### PR TITLE
Added correct initial state for cart and wishlist

### PR DESCRIPTION
**Pull request by dmytruk-y**
## Description

Added correct initial state for cart and wishlist. Also removed redundant code.

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/redux/common-store/common.reducer.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import { SET_CART , SET_WISHLIST } from './common.types';
+import { SET_CART, SET_WISHLIST } from './common.types';
+import { getFromLocalStorage } from '../../services/local-storage.service';
+import { CART_KEY, WISHLIST_KEY } from '../../configs';
 
 export const initialState = {
-  cart: [],
-  wishlist: []
+  cart: getFromLocalStorage(CART_KEY),
```
Comment (maarekua) [inline src/redux/common-store/common.reducer.js:6]: The initial state behind the component should be a constant. You should use the lifecycle hook if you want to set it from the local storage on the mount.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/redux/common-store/common.reducer.js (hunk 1 above)
Comment (dmytruk-y) [reply src/redux/common-store/common.reducer.js]: Fixed. Now the state is updated with useEffect

File: src/containers/cart-header/cart-header.js [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
-  }));
   const styles = useStyles({ fromSideBar });
+  const dispatch = useDispatch();
+  const cartItems = useSelector(({ CommonStore }) => CommonStore.cart);
   const itemsCount = useMemo(
     () => cartItems.length && cartItems.reduce((acc, item) => acc + item.quantity, 0),
     [cartItems]
   );
+  useEffect(() => {
+    dispatch(setCart(getFromLocalStorage(CART_KEY)));
```
Comment (LubomyrKashuba) [inline src/containers/cart-header/cart-header.js:24]: we have tried to avoid using redux. Use useCart hook for manage cart

**Review by tarasSanotskyi — APPROVED** [review]

File: src/containers/checkout/checkout-form/tests/checkout-form.spec.js [hunk 3]
```diff
@@ -109,70 +109,81 @@ describe('CheckoutForm tests for: ', () => {
...
   });
-  it('first name field', () => {
-    const firstNameField = screen.getByTestId('firstName').querySelector('input');
-    fireEvent.change(firstNameField, { target: { value: 'Roman' } });
-    expect(firstNameField.value).toEqual('Roman');
+  it('first name field', async () => {
+    const firstNameField = await screen.findByTestId('firstName');
+    const input = firstNameField.querySelector('input');
+    fireEvent.change(input, { target: { value: 'Roman' } });
+    expect(input.value).toEqual('Roman');
```
Comment (tarasSanotskyi) [inline src/containers/checkout/checkout-form/tests/checkout-form.spec.js:116]: add empty lines before all 'expect' for better code readability

**Review by tarasSanotskyi — CHANGES_REQUESTED** [review]

File: src/containers/cart-header/cart-header.js (hunk 2 above)
Comment (dmytruk-y) [reply src/containers/cart-header/cart-header.js:24]: Without using the redux store I cannot get the badge updates from other components. I will try to find out if there is some good solution to this issue.

File: src/containers/checkout/checkout-form/tests/checkout-form.spec.js (hunk 3 above)
Comment (dmytruk-y) [reply src/containers/checkout/checkout-form/tests/checkout-form.spec.js:117]: Done.

**Review by maarekua — APPROVED** [review]

**Review by tarasSanotskyi — APPROVED** [review]

File: src/containers/cart-header/cart-header.js (hunk 2 above)
Comment (dmytruk-y) [reply src/containers/cart-header/cart-header.js]: Rewrote it using the React Context.

**Review by maarekua — APPROVED** [review]

**Review by MickhaelSh — APPROVED** [review]

**Pull request merged**